File: 2Lab/labs6/lab6.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cctype>

class IDigitProcessor {
public:
    virtual int CountDigits() const = 0;
    virtual std::string RemoveDigits() const = 0;
    virtual ~IDigitProcessor() = default;
};

class TextLine : public IDigitProcessor {
private:
    std::string content;

public:
    TextLine(const std::string& text) : content(text) {}

    std::string GetContent() const {
        return content;
    }

    int CountDigits() const override {
        int count = 0;
        for (char c : content) {
            if (isdigit(c)) ++count;
        }
        return count;
    }

    std::string RemoveDigits() const override {
        std::string result;
        for (char c : content) {
            if (!isdigit(c)) result += c;
        }
        return result;
    }
};

class TextContainer {
private:
    std::vector<TextLine> lines;

public:
    void AddLine(const TextLine& line) {
        lines.push_back(line);
    }

    void Clear() {
        lines.clear();
    }

    int TotalDigitCount() const {
        int total = 0;
        for (const auto& line : lines) {
            total += line.CountDigits();
        }
        return total;
    }
};

int main() {
    TextContainer container;
    container.AddLine(TextLine("Line1 with 123"));
    container.AddLine(TextLine("Another456Line"));

    std::cout << "Total digits: " << container.TotalDigitCount() << std::endl;
    return 0;
}

File: 2Lab/labs6/lab6.cs
using System;
using System.Collections.Generic;
using System.Linq;

public interface IDigitProcessor {
    int CountDigits();
    string RemoveDigits();
}

public class TextLine : IDigitProcessor {
    private readonly string content;

    public TextLine(string text) {
        content = text;
    }

    public string GetContent() => content;

    public int CountDigits() {
        return content.Count(char.IsDigit);
    }

    public string RemoveDigits() {
        return new string(content.Where(c => !char.IsDigit(c)).ToArray());
    }
}

public class TextContainer {
    private readonly List<TextLine> lines = new();

    public void AddLine(TextLine line) => lines.Add(line);

    public void Clear() => lines.Clear();

    public int TotalDigitCount() => lines.Sum(line => line.CountDigits());
}

class Program {
    static void Main() {
        var container = new TextContainer();
        container.AddLine(new TextLine("Example123Text"));
        container.AddLine(new TextLine("Another456Example"));

        Console.WriteLine("Total digits: " + container.TotalDigitCount());
    }
}
